app/include/MainController.hpp

```diff
@@ -0,0 +1,58 @@
+//
+// Created by marko on 2/5/25.
+//
+
+#ifndef MAINCONTROLLER_HPP
+#define MAINCONTROLLER_HPP
+#include <engine/core/Controller.hpp>
+#include <glm/glm.hpp>
+#include "Framebuffer.hpp"
+#include "MoonEvent.hpp"
+
+struct Light{
+    glm::vec3 intensity = glm::vec3(1.4f);
+    glm::vec3 position = glm::vec3(15.f, 30.f, -20.f);
+    glm::vec3 ambient = glm::vec3(0.8f);
+    glm::vec3 diffuse = glm::vec3(1.0f);
+    glm::vec3 specular = glm::vec3(1.0f);
+};
+
+struct SpotLight {
+    glm::vec3 diffuse = glm::vec3(0.9f);
+    glm::vec3 specular = glm::vec3(0.9f);
+    float cut_off = cos(glm::radians(12.5f));
+    float outterCut_off = cos(glm::radians(17.5f));
+};
+
+class MainController : public engine::core::Controller {
+    friend class MoonEvent;
+    friend class MainPlatformEventObserver;
+public:
+    MoonEvent* moon_event_handler;
+    float exposure = 1;
+    glm::vec3 spacecraft_pos = glm::vec3(0.0f, -2.0f, -30.0f);
+    float spacecraft_rotation = glm::radians(0.f);
```

Comment:
Da li se ovim poljima zaista pristupa iz neke druge klase spolja? Ako je to slučaj, možda vredi razmotriti da li je taj pristup zaista neophodan. Ako nije, možda bi bilo bolje postaviti ih kao private radi bolje enkapsulacije.